src/components/CreateNewPin.js

```diff
@@ -8,25 +13,178 @@ import {
   CreatePinButton,
   UploadNewPinImageLabel,
   UploadNewPinImageInput,
+  PreviewImage,
 } from "../styles/CreateNewPin.module";
+import imageCompression from "browser-image-compression";
+
+const firebaseConfig = {
+  // eslint-disable-next-line no-undef
+  apiKey: process.env.REACT_APP_FIREBASE_APIKEY,
+  authDomain: process.env.REACT_APP_FIREBASE_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECTID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGEBUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGINGSENDERID,
+  appId: process.env.REACT_APP_FIREBASE_APPID,
+  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENTID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
 
 function CreateNewPin() {
-  const [pinName, setPinName] = useState();
-  const [pinDescription, setPinDescription] = useState();
-  const [pinLink, setPinLink] = useState();
+  const [pinName, setPinName] = useState("");
+  const [pinDescription, setPinDescription] = useState("");
+  const [pinLink, setPinLink] = useState("");
+  const [pinImage, setPinImage] = useState();
+  const [isGetPinImage, setIsGetPinImage] = useState(false);
+  const [selectedFile, setSelectedFile] = useState();
+  const [preview, setPreview] = useState();
+  const [isPinCreated, setIsPinCreated] = useState(false);
+
+  const redirect = useNavigate();
+
+  async function handleImageUpload(e) {
+    const imageFile = e.target.files[0];
```

Comment:
I think it will cause some uncaught error if 'e.target.files' is undefined. Since we cannot get the '0' index from undefined.
You could try to declare this file after lines 51-55